src/main/java/io/jawg/osmcontributor/ui/activities/PhotoActivity.java

```diff
@@ -217,9 +217,7 @@ protected void onCreate(Bundle savedInstanceState) {
         // Init parameters.
         latitude = getIntent().getDoubleExtra("latitude", 0);
         longitude = getIntent().getDoubleExtra("longitude", 0);
-        poiId = getIntent().getLongExtra("poiId", 0);
-        imageAdapter = new ImageAdapter(this, poiId);
-        gridPhotos.setAdapter(imageAdapter);
+        imageAdapters = new HashMap<Long, ImageAdapter>();
```

Comment:
Yes. Basically, you could filter-out the results by poiId instead of keeping multiple adapters.